File: apps/board/web/src/types/meeting.ts
export interface MeetingAttendee {
  email?: string;
  name?: string;
  status?: string;
}

export interface Meeting {
  id: string;
  external_id: string | null;
  title: string;
  description: string | null;
  starts_at: string;
  ends_at: string;
  location: string | null;
  meet_link: string | null;
  attendees: MeetingAttendee[];
  contact_id: string | null;
  lead_id: string | null;
  created_at: string;
  updated_at: string;
}

export interface MeetingCreatePayload {
  title: string;
  description?: string | null;
  starts_at: string;
  ends_at: string;
  location?: string | null;
  meet_link?: string | null;
  attendees?: MeetingAttendee[];
  contact_id?: string | null;
  lead_id?: string | null;
}

export type MeetingUpdatePayload = Partial<MeetingCreatePayload>;

File: apps/board/backend/app/models/meeting.py
from __future__ import annotations

from datetime import datetime
from typing import Any, List, Optional
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field


class MeetingBase(BaseModel):
    external_id: Optional[str] = None
    title: str
    description: Optional[str] = None
    starts_at: datetime
    ends_at: datetime
    location: Optional[str] = None
    meet_link: Optional[str] = None
    attendees: List[Any] = Field(default_factory=list)
    contact_id: Optional[UUID] = None
    lead_id: Optional[UUID] = None


class MeetingCreate(MeetingBase):
    pass


class MeetingUpdate(BaseModel):
    external_id: Optional[str] = None
    title: Optional[str] = None
    description: Optional[str] = None
    starts_at: Optional[datetime] = None
    ends_at: Optional[datetime] = None
    location: Optional[str] = None
    meet_link: Optional[str] = None
    attendees: Optional[List[Any]] = None
    contact_id: Optional[UUID] = None
    lead_id: Optional[UUID] = None


class MeetingRead(MeetingBase):
    model_config = ConfigDict(from_attributes=True)
    id: UUID
    created_at: datetime
    updated_at: datetime
